tools/nblIntrinicGenerator/example input/spirv.type_mappings.json

```diff
@@ -0,0 +1,59 @@
+{
+    "default": "uint32_t",
+    "classMappings":[
+        {
+            "class": "Atomic",
+            "types": [
+                {
+                    "kind": "IdResult",
+                    "type": "T"
+                },
```

Comment:
we should probably call `type`, `cpp_type` or `hlsl_type` instead

and apply it to `IdResultType`

Reply:
hmm actually, you should have `hlsl_type` e.g. `const T&` -> this means constant, ext_reference

but also list generics separately (once for the whole function).

The reason is that sometimes you might end up with 4 template dependant parameters (if `vk::SpirvType<>` gets involved), and only 2 template args

Reply:
I might add an optional field that allows the user to list generic typenames that dont fit in the standard "single alphabetic char, uppercase" , and then merge attributes, const and type into `hlsl_type` field.